tests/tests/optionsTests.robot

```diff
@@ -124,4 +125,96 @@ Verify Output log to a file option
     Click Load Button
     Wait Until Keyword Succeeds  5s  1s  Click Element  ${DownloadLogsButton}
     Switch Window    new
-    Wait Until Element Is Visible    ${OpeningFileText}
\ No newline at end of file
+    Wait Until Element Is Visible    ${OpeningFileText}
+
+Verify Extraction filter set to 'Chosen language and language agnostic files'
+    [Documentation]    Chosen language and language agnostic files extraction filter option
+    [Tags]    option    extraction filter
+
+    ${downloaded_file_path}    Set Variable    ${DOWNLOAD_PATH}${extraction_filter}[archive_name].zip
+    Click Element    ${OptionsButton}
+    Wait Until Element Is Visible    ${OptionsList}
+    Click Add Files Button
+    Upload Test File    ${extraction_filter}[path]
+    Select From List By Index    ${ExtractionLangFilterOption}    0
+    List Selection Should Be     ${ExtractionLangFilterOption}    lang-plus-agn
+    Select From List By Index    ${CollisionResolutionOption}    1
+    List Selection Should Be     ${CollisionResolutionOption}    rename
+    Click Load Button
+    Click Extract And Save Button    ${extraction_timeout}
+    Wait Until Created    ${downloaded_file_path}
+    Validate and Unzip Test File    ${downloaded_file_path}
+    ${ListFiles}  List Files In Directory   ${DOWNLOAD_PATH}${extraction_filter}[archive_name]/app
+    Should Be Equal As Strings     ${ListFiles}    ['MyProg.chm', 'MyProg.exe', 'Readme.txt']
+    Remove File    ${downloaded_file_path}
+    Remove Directory    ${DOWNLOAD_PATH}${extraction_filter}[archive_name]    True
```

Comment:
It might be useful to move last two lines to "test teardown" so these lines will be run even if test fails